synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -367,7 +368,128 @@ async def _do_preview(self, url: str, user: UserID, ts: int) -> bytes:
 
         return jsonog.encode("utf8")
 
-    async def _download_url(self, url: str, user: UserID) -> MediaInfo:
+    async def _download_url(
+        self, url: str, output_stream: BinaryIO
+    ) -> Tuple[int, str, int, str, Optional[str], int, Optional[str]]:
```

Comment:
Maybe this should be another type instead of just a massive tuple? 🤷

Reply:
Yeeeees, that'd be nice, please do :-)